### ISSUE TITLE
DL4J error when running with fat jar

### ISSUE THREAD

**Issue by bgyori** [clulab/eidos#486]
When running extractFromDirectory via an assembled fat jar, I'm getting this error:
```
18:12:13.419 [main] WARN  o.d.n.m.keras.utils.KerasModelUtils - Could not read keras backend used (no backend field found) 

18:12:13.448 [main] WARN  o.d.n.m.k.l.e.KerasEmbedding - Masking in keras and DL4J work differently. We do not completely support mask_zero flag on Embedding layers. Zero Masking for the Embedding layer only works with unidirectional LSTM for now. If you want to have this behaviour for your imported model in DL4J, apply masking as a pre-processing step to your input.See https://deeplearning4j.org/usingrnns#masking for more on this.
Exception in thread "main" java.lang.ExceptionInInitializerError
	at org.deeplearning4j.nn.conf.dropout.Dropout.initializeHelper(Dropout.java:107)
	at org.deeplearning4j.nn.conf.dropout.Dropout.<init>(Dropout.java:100)
	at org.deeplearning4j.nn.conf.dropout.Dropout.<init>(Dropout.java:80)
	at org.deeplearning4j.nn.conf.layers.Layer$Builder.dropOut(Layer.java:257)
	at org.deeplearning4j.nn.modelimport.keras.layers.embeddings.KerasEmbedding.<init>(KerasEmbedding.java:125)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasLayerUtils.getKerasLayerFromConfig(KerasLayerUtils.java:258)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.prepareLayers(KerasModel.java:220)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:166)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:98)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildModel(KerasModelBuilder.java:305)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasModelAndWeights(KerasModelImport.java:127)
	at org.clulab.timenorm.TemporalCharbasedParser.<init>(TemporalCharbasedParser.scala:56)
	at org.clulab.wm.eidos.EidosSystem$LoadableAttributes$.apply(EidosSystem.scala:164)
	at org.clulab.wm.eidos.EidosSystem.<init>(EidosSystem.scala:190)
	at org.clulab.wm.eidos.apps.ExtractFromDirectory$.delayedEndpoint$org$clulab$wm$eidos$apps$ExtractFromDirectory$1(ExtractFromDirectory.scala:13)
	at org.clulab.wm.eidos.apps.ExtractFromDirectory$delayedInit$body.apply(ExtractFromDirectory.scala:9)
	at scala.Function0.apply$mcV$sp(Function0.scala:34)
	at scala.Function0.apply$mcV$sp$(Function0.scala:34)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
	at scala.App.$anonfun$main$1$adapted(App.scala:76)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at scala.App.main(App.scala:76)
	at scala.App.main$(App.scala:74)
	at org.clulab.wm.eidos.apps.ExtractFromDirectory$.main(ExtractFromDirectory.scala:9)
	at org.clulab.wm.eidos.apps.ExtractFromDirectory.main(ExtractFromDirectory.scala)
Caused by: java.lang.RuntimeException: org.nd4j.linalg.factory.Nd4jBackend$NoAvailableBackendException: Please ensure that you have an nd4j backend on your classpath. Please see: http://nd4j.org/getstarted.html
	at org.nd4j.linalg.factory.Nd4j.initContext(Nd4j.java:5449)
	at org.nd4j.linalg.factory.Nd4j.<clinit>(Nd4j.java:213)
	... 25 more
Caused by: org.nd4j.linalg.factory.Nd4jBackend$NoAvailableBackendException: Please ensure that you have an nd4j backend on your classpath. Please see: http://nd4j.org/getstarted.html
	at org.nd4j.linalg.factory.Nd4jBackend.load(Nd4jBackend.java:213)
	at org.nd4j.linalg.factory.Nd4j.initContext(Nd4j.java:5446)
	... 26 more
```

to reproduce, try
```
sbt -J-Xmx8G assembly
java -Xmx8G -cp target/scala-2.12/eidos-assembly-0.2.2-SNAPSHOT.jar org.clulab.wm.eidos.apps.ExtractFromDirectory /path/to/input /path/to/output
```
Also note that useTimeNorm is set to true in eidos.conf. Any ideas what the problem could be? I didn't have this issue a couple of weeks ago when I built the system.

**Comment by EgoLaparra** [comment]
Are you running the jar in the same system you built it?

**Comment by bgyori** [comment]
Yes, using the same system

**Comment by kwalcock** [comment]
I also get the error if I do the assembly step rather than run straight from sbt.

**Comment by kwalcock** [comment]
@EgoLaparra, I'm hoping that you have some insight into this.  Does grepping for "Masking in keras and DL4J work differently. We do not completely support mask_zero flag on" in the source code help?

**Comment by EgoLaparra** [comment]
I'm afraid that sbt is having problems with some transitive dependency when assembling the fat jar. I'm trying to package eidos as a maven project to check this.

**Comment by kwalcock** [comment]
You probably noticed that eidos is using
```addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.6")```
while the current version has been updated to 0.14.8.

I did recently change this code in build.sbt:
```
libraryDependencies ++= {
  val (major, minor) = CrossVersion.partialVersion(scalaVersion.value).get
  val timenorm = "timenorm-0.9.6.15" + (if (minor == 11) "_2.11.11" else "")

  Seq("com.github.clulab" % "timenorm" % timenorm exclude("org.slf4j", "slf4j-log4j12"))
}
```

**Comment by kwalcock** [comment]
It looks like I should have "if (minor.toInt == 11)".  Are you using Scala 2.11.11, @bgyori?

**Comment by bgyori** [comment]
I have 2.12.4 on my Mac and 2.11.8 on my Debian

**Comment by EgoLaparra** [comment]
So, the jar built with maven works. I will try with sbt-onejar instead of sbt-assembly to check if the plugin is the problem.

**Comment by kwalcock** [comment]
Good work, everyone.
